django/db/backends/base/schema.py

```diff
@@ -223,7 +223,11 @@ def column_sql(self, model, field, include_default=False):
             default_value = self.effective_default(field)
             column_default = ' DEFAULT ' + self._column_default_sql(field)
             if default_value is not None:
-                if self.connection.features.requires_literal_defaults:
+                if hasattr(default_value, 'as_sql'):
+                    s, p = default_value.as_sql(self, self.connection)
```

Comment:
Use `sql, params =` instead of `s, p =` to match the typical naming used elsewhere in the code.